gemma-core/src/main/java/ubic/gemma/model/expression/experiment/ExpressionExperiment.java

```diff
@@ -198,7 +199,7 @@ public void setBatchEffect( String batchEffect ) { // FIXME don't use a string f
     @Override
     public void setBioAssays( Collection<BioAssay> bioAssays ) {
         this.bioAssays = bioAssays;
-        if ( bioAssays != null )
+        if ( bioAssays != null && Hibernate.isInitialized( bioAssays ) )
```

Comment:
We should always check for `Hibernate.isInitialized` in the model layer because accessing properties like this can inadvertently query the database. If we need the relationship, we can use `Hibernate.initialize` or ensure it's there with a fetch jointure.